### PR TITLE
Fix Xapian query regression in travel app

### PR DESCRIPTION
**Pull request by ptomato**
https://phabricator.endlessm.com/T17143

### CODE REVIEW

File: js/app/modules/selection/xapian.js [hunk 1]
```diff
@@ -56,6 +56,14 @@ const Xapian = new Module.Class({
         throw new Error('You should be implementing construct_query_object in your subclass');
     },
 
+    _modify_query_object(query) {
```
Comment (tchx84) [inline js/app/modules/selection/xapian.js]:
```
Gjs-CRITICAL **: JS ERROR: SyntaxError: missing : after property id @ resource:///com/endlessm/knowledge/js/app/modules/selection/xapian.js:59

```

**Comment by tchx84** [comment]
@ptomato I tried this branch, and even if I fix the previous comment myself, I still get the following errors, but I can't make sense of it:

```
(com.endlessm.travel.en:32162): Gjs-WARNING **: JS ERROR: TypeError: subs is undefined
Application<._check_for_update@resource:///com/endlessm/knowledge/js/app/application.js:199:9
wrapper@resource:///org/gnome/gjs/modules/lang.js:178:13
Application<._check_for_content@resource:///com/endlessm/knowledge/js/app/application.js:172:9
wrapper@resource:///org/gnome/gjs/modules/lang.js:178:13
Application<.vfunc_startup@resource:///com/endlessm/knowledge/js/app/application.js:219:9
wrapper@resource:///org/gnome/gjs/modules/lang.js:178:13
@<command line>:10:18


(com.endlessm.travel.en:32162): Gjs-WARNING **: JS ERROR: TypeError: Eknc.QueryObject.new_from_props is not a function
AllSets<.construct_query_object@resource:///com/endlessm/knowledge/js/app/modules/selection/allSets.js:27:13
wrapper@resource:///org/gnome/gjs/modules/lang.js:178:13
Xapian<.queue_load_more@resource:///com/endlessm/knowledge/js/app/modules/selection/xapian.js:75:13
wrapper@resource:///org/gnome/gjs/modules/lang.js:178:13
ContentGroup<.load@resource:///com/endlessm/knowledge/js/app/modules/contentGroup/contentGroup.js:346:9
wrapper@resource:///org/gnome/gjs/modules/lang.js:178:13
ContentGroup<._ensure_synced@resource:///com/endlessm/knowledge/js/app/modules/contentGroup/contentGroup.js:233:13
wrapper@resource:///org/gnome/gjs/modules/lang.js:178:13
ContentGroup<._init@resource:///com/endlessm/knowledge/js/app/modules/contentGroup/contentGroup.js:207:9
Class<._construct/props._init@resource:///com/endlessm/knowledge/js/app/knowledge.js:76:17
wrapper@resource:///org/gnome/gjs/modules/lang.js:178:13
ModuleFactory<._create_module@resource:///com/endlessm/knowledge/js/app/moduleFactory.js:124:13
wrapper@resource:///org/gnome/gjs/modules/lang.js:178:13
ModuleFactory<.create_module_for_slot@resource:///com/endlessm/knowledge/js/app/moduleFactory.js:213:13
wrapper@resource:///org/gnome/gjs/modules/lang.js:178:13
Module<.create_submodule@resource:///com/endlessm/knowledge/js/app/interfaces/module.js:203:1
Scrolling<._init@resource:///com/endlessm/knowledge/js/app/modules/layout/scrolling.js:30:9
Class<._construct/props._init@resource:///com/endlessm/knowledge/js/app/knowledge.js:76:17
wrapper@resource:///org/gnome/gjs/modules/lang.js:178:13
ModuleFactory<._create_module@resource:///com/endlessm/knowledge/js/app/moduleFactory.js:124:13
wrapper@resource:///org/gnome/gjs/modules/lang.js:178:13
ModuleFactory<.create_module_for_slot@resource:///com/endlessm/knowledge/js/app/moduleFactory.js:213:13
wrapper@resource:///org/gnome/gjs/modules/lang.js:178:13
Module<.create_submodule@resource:///com/endlessm/knowledge/js/app/interfaces/module.js:203:1
SideMenu<._init@resource:///com/endlessm/knowledge/js/app/modules/layout/sideMenu.js:90:13
Class<._construct/props._init@resource:///com/endlessm/knowledge/js/app/knowledge.js:76:17
wrapper@resource:///org/gnome/gjs/modules/lang.js:178:13
ModuleFactory<._create_module@resource:///com/endlessm/knowledge/js/app/moduleFactory.js:124:13
wrapper@resource:///org/gnome/gjs/modules/lang.js:178:13
ModuleFactory<.create_module_for_slot@resource:///com/endlessm/knowledge/js/app/moduleFactory.js:213:13
wrapper@resource:///org/gnome/gjs/modules/lang.js:178:13
Module<.create_submodule@resource:///com/endlessm/knowledge/js/app/interfaces/module.js:203:1
Simple<._init@resource:///com/endlessm/knowledge/js/app/modules/pager/simple.js:55:9
Class<._construct/props._init@resource:///com/endlessm/knowledge/js/app/knowledge.js:76:17
wrapper@resource:///org/gnome/gjs/modules/lang.js:178:13
ModuleFactory<._create_module@resource:///com/endlessm/knowledge/js/app/moduleFactory.js:124:13
wrapper@resource:///org/gnome/gjs/modules/lang.js:178:13
ModuleFactory<.create_module_for_slot@resource:///com/endlessm/knowledge/js/app/moduleFactory.js:213:13
wrapper@resource:///org/gnome/gjs/modules/lang.js:178:13
Module<.create_submodule@resource:///com/endlessm/knowledge/js/app/interfaces/module.js:203:1
MediaLightbox<._init@resource:///com/endlessm/knowledge/js/app/modules/contentGroup/mediaLightbox.js:36:9
Class<._construct/props._init@resource:///com/endlessm/knowledge/js/app/knowledge.js:76:17
wrapper@resource:///org/gnome/gjs/modules/lang.js:178:13
ModuleFactory<._create_module@resource:///com/endlessm/knowledge/js/app/moduleFactory.js:124:13
wrapper@resource:///org/gnome/gjs/modules/lang.js:178:13
ModuleFactory<.create_module_for_slot@resource:///com/endlessm/knowledge/js/app/moduleFactory.js:213:13
wrapper@resource:///org/gnome/gjs/modules/lang.js:178:13
Module<.create_submodule@resource:///com/endlessm/knowledge/js/app/interfaces/module.js:203:1
BrandPage<._init@resource:///com/endlessm/knowledge/js/app/modules/layout/brandPage.js:39:9
Class<._construct/props._init@resource:///com/endlessm/knowledge/js/app/knowledge.js:76:17
wrapper@resource:///org/gnome/gjs/modules/lang.js:178:13
ModuleFactory<._create_module@resource:///com/endlessm/knowledge/js/app/moduleFactory.js:124:13
wrapper@resource:///org/gnome/gjs/modules/lang.js:178:13
ModuleFactory<.create_module_for_slot@resource:///com/endlessm/knowledge/js/app/moduleFactory.js:213:13
wrapper@resource:///org/gnome/gjs/modules/lang.js:178:13
Module<.create_submodule@resource:///com/endlessm/knowledge/js/app/interfaces/module.js:203:1
Simple<._init@resource:///com/endlessm/knowledge/js/app/modules/window/simple.js:66:9
Class<._construct/props._init@resource:///com/endlessm/knowledge/js/app/knowledge.js:76:17
wrapper@resource:///org/gnome/gjs/modules/lang.js:178:13
ModuleFactory<._create_module@resource:///com/endlessm/knowledge/js/app/moduleFactory.js:124:13
wrapper@resource:///org/gnome/gjs/modules/lang.js:178:13
ModuleFactory<.create_module_for_slot@resource:///com/endlessm/knowledge/js/app/moduleFactory.js:213:13
wrapper@resource:///org/gnome/gjs/modules/lang.js:178:13
Module<.create_submodule@resource:///com/endlessm/knowledge/js/app/interfaces/module.js:203:1
Buffet<._init@resource:///com/endlessm/knowledge/js/app/modules/controller/buffet.js:37:13
Class<._construct/props._init@resource:///com/endlessm/knowledge/js/app/knowledge.js:76:17
wrapper@resource:///org/gnome/gjs/modules/lang.js:178:13
ModuleFactory<._create_module@resource:///com/endlessm/knowledge/js/app/moduleFactory.js:124:13
wrapper@resource:///org/gnome/gjs/modules/lang.js:178:13
ModuleFactory<.create_root_module@resource:///com/endlessm/knowledge/js/app/moduleFactory.js:158:1
wrapper@resource:///org/gnome/gjs/modules/lang.js:178:13
Application<._ensure_controller@resource:///com/endlessm/knowledge/js/app/application.js:281:13
wrapper@resource:///org/gnome/gjs/modules/lang.js:178:13
Application<.vfunc_activate@resource:///com/endlessm/knowledge/js/app/application.js:244:9
wrapper@resource:///org/gnome/gjs/modules/lang.js:178:13
@<command line>:10:18
```

I tried com.endlessm.travel.en in master and don't see those errors.

File: js/app/modules/selection/xapian.js (hunk 1 above)
Comment (ptomato) [inline js/app/modules/selection/xapian.js]: Are you running the system 1.47.x version of GJS? This syntax is allowed in 1.48 which is what we shipped with our runtime. We should be using it for new code.

**Comment by ptomato** [comment]
Try jhbuilding mozjs38 and gjs from gnome's 3.24 moduleset, that will match what we have in our runtime.

**Pull request merged**